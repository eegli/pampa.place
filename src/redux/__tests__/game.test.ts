import gameReducer, { gameSlice, initialState } from '../slices/game';

export const {
  reset,
  initGame,
  startRound,
  finishRound,
  resetRound,
  setPlayerScore,
  setRounds,
  setMap,
  setPlayers,
  setTimeLimit,
} = gameSlice.actions;

describe('Redux, game state', () => {
  it('creates game with default players if none are present', () => {
    const state = initialState;
    expect(gameReducer(state, initGame)).toMatchSnapshot();
  });
});
