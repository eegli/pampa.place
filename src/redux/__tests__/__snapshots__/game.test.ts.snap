// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux, game state creates game with default players if none are present 1`] = `
Object {
  "mapId": "eJmFZpriK7QX",
  "mapName": "Alpen",
  "players": Object {
    "names": Array [
      "Player 1",
    ],
    "scores": Object {
      "Player 1": Object {
        "results": Array [],
        "totalScore": 0,
      },
    },
  },
  "rounds": Object {
    "current": 1,
    "progress": 0,
    "total": 3,
  },
  "status": "INTERMISSION",
  "timeLimit": 60,
}
`;
